viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -647,6 +647,23 @@ void SLAMServiceImpl::SaveMapWithTimestamp() {
     }
 }
 
+cartographer::transform::Rigid3d SLAMServiceImpl::AddSensorReading(
+    cartographer::sensor::TimedPointCloudData measurement,
+    cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder,
+    int trajectory_id, cartographer::transform::Rigid3d tmp_global_pose) {
+    std::lock_guard<std::mutex> lk(map_builder_mutex);
+
+    if (measurement.ranges.size() > 0) {
+        trajectory_builder->AddSensorData(kRangeSensorId.id, measurement);
+        auto local_poses = map_builder.GetLocalSlamResultPoses();
+        if (local_poses.size() > 0) {
+            tmp_global_pose =
+                map_builder.GetGlobalPose(trajectory_id, local_poses.back());
+        }
+    }
+    return tmp_global_pose;
```

Comment:
[nit] This function should ideally return `void`, and `latest_global_pose` should be set within this function since updating it goes hand in hand with adding a new sensor reading.